controllers/jenkins/pipelinerun/pipelinerun_controller.go

```diff
@@ -186,6 +200,50 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	return ctrl.Result{RequeueAfter: 1 * time.Second}, nil
 }
 
+func (r *Reconciler) deleteJenkinsJobHistory(pipelineRun *prv1alpha3.PipelineRun) (err error) {
+	var buildNum int
+	if buildNum = getJenkinsBuildNumber(pipelineRun); buildNum < 0 {
+		return
+	}
+
+	jenkinsClient := job.Client{
+		JenkinsCore: r.JenkinsCore,
+	}
+	jobName := fmt.Sprintf("job/%s/job/%s", pipelineRun.Namespace, pipelineRun.Spec.PipelineRef.Name)
+	if err = jenkinsClient.DeleteHistory(jobName, buildNum); err == nil {
+		pipelineRun.ObjectMeta.Finalizers = sliceutil.RemoveString(pipelineRun.ObjectMeta.Finalizers, func(item string) bool {
```

Comment:
IMO, the finalizer removal should be outside this function. This function should keep simplicity and only deletes the external resource.